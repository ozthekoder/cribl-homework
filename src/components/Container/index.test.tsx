import { render } from "@testing-library/react";
import { Container } from ".";
import "@testing-library/jest-dom";

describe("Container Component", () => {
  test("should render correctly with default styles", () => {
    const { container } = render(<Container>Test Content</Container>);

    expect(container.firstChild).toHaveStyle({
      height: "100%",
      width: "100%",
      overflowY: "auto",
      position: "relative",
    });
  });

  test("should render children correctly", () => {
    const { getByText } = render(<Container>Child Element</Container>);

    expect(getByText("Child Element")).toBeInTheDocument();
  });

  test("should match snapshot", () => {
    const { container } = render(<Container>Snapshot Test</Container>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
