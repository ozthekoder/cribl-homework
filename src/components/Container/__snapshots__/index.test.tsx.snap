// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component should match snapshot 1`] = `
<div
  class="css-10b489u"
>
  Snapshot Test
</div>
`;
